internal/dot/graph.go

```diff
@@ -42,3 +44,30 @@ type Node struct {
 	Optional bool
 	Group    string
 }
+
+// String returns the string representation of a node so the different
+// constructors can refer to the same node. We omit information on the optional
+// field since the same type can be optional to one constructor and required
+// for another.
+func (n *Node) String() string {
```

Comment:
nit: Are we doing anything different here than just implementing the
`Stringer` interface? If not, the doc for this can just be,

    // String implements fmt.Stringer for Node.
    
Or,

    // String returns a user-readable string representation of Node.

Reply:
We might want to explain why we omit the `optional` field in the string representation since the `String()` method for `paramSingle` includes that.